src/main/java/com/theelixrlabs/healthcare/service/PatchDoctorService.java

```diff
@@ -0,0 +1,72 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.DoctorConstants;
+import com.theelixrlabs.healthcare.constants.MessageConstants;
+import com.theelixrlabs.healthcare.dto.DoctorDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.model.DoctorModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.validation.Validator;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Service class for facilitating partial updates to doctor details.
+ */
+@Service
+public class PatchDoctorService {
+
+    private final DoctorRepository doctorRepository;
+
+    public PatchDoctorService(DoctorRepository doctorRepository) {
+        this.doctorRepository = doctorRepository;
+    }
+
+    /**
+     * Replace existing doctor details based on request.
+     *
+     * @param doctorId  Doctor unique UUID
+     * @param doctorDto Data transfer object containing doctor information.
+     * @return The replaced doctor details.
+     * @throws CustomException If validation fails or doctor already exists.
+     */
+    public DoctorDto patchDoctorById(UUID doctorId, DoctorDto doctorDto) throws CustomException {
+        Validator doctorValidator = new Validator();
+        doctorValidator.validateDoctor(doctorDto);
+        Optional<DoctorModel> optionalDoctor = doctorRepository.findById(doctorId);
+        if (optionalDoctor.isPresent()) {
+            DoctorModel existingDoctor = optionalDoctor.get();
+            if (doctorDto.getFirstName() != null && !doctorDto.getFirstName().equals(existingDoctor.getFirstName())) {
+                existingDoctor.setFirstName(doctorDto.getFirstName());
+            }
+            if (doctorDto.getLastName() != null && !doctorDto.getLastName().equals(existingDoctor.getLastName())) {
+                existingDoctor.setLastName(doctorDto.getLastName());
+            }
+            if (doctorDto.getDepartment() != null && !doctorDto.getDepartment().equals(existingDoctor.getDepartment())) {
```

Comment:
I feel this will fail in the scenario when dto and existing is having same values , have u checked ?
I suggest to do change as below
```suggestion
            if (doctorDto.getDepartment() != null) {
                            existingDoctor.setDepartment(!doctorDto.getDepartment().equals(existingDoctor.getDepartment() ? 
                            existingDoctor.getDepartment() : doctorDto.getDepartment());
            }
```